source/contracts/Controller.sol

```diff
@@ -8,9 +8,17 @@ import 'libraries/Extractable.sol';
 
 
 contract Controller is IController {
+    struct ContractDetails {
+        bytes32 name;
```

Comment:
Why store the name here?

Reply:
My thought was that now that Solidity supports returning/passing structs between functions that we may use this structure to pass contract details around in the future.  By putting the name into the struct, this allows us to have a single object that contains all of the relevant details since you may not always get one of these structures by looking it up in the mapping directly.

If you think this is too much of a premature optimization I don't mind removing it.  I don't care much about the increased gas costs since this cost is only incurred at contract upload time, where we really don't care at all about gas costs.

Reply:
Sounds good! Was just curious on the intent since it isn't used yet.